File: app/src/main/java/im/tox/antox/Message.java
package im.tox.antox;

import java.sql.Timestamp;

/**
 * Created by ollie on 10/03/14.
 */
public class Message {
    public int message_id;
    public String key;
    public String message;
    public boolean is_outgoing;
    public boolean has_been_received;
    public boolean has_been_read;
    public Timestamp timestamp;

    public Message() {
        super();
    }

    public Message(int message_id, String k, String m, boolean outgoing, boolean received, boolean read, Timestamp time) {
        super();
        this.message_id = message_id;
        this.key = k;
        this.message = m;
        this.is_outgoing = outgoing;
        this.has_been_received = received;
        this.has_been_read = read;
        this.timestamp = time;
    }

}

File: app/src/main/scala/chat/tox/antox/wrapper/Message.scala
package chat.tox.antox.wrapper

import java.sql.Timestamp

import android.content.Context
import chat.tox.antox.R
import chat.tox.antox.data.CallEventKind
import chat.tox.antox.tox.ToxSingleton
import chat.tox.antox.utils.{FileUtils, TimestampUtils}
import chat.tox.antox.wrapper.MessageType.MessageType

case class Message(id: Int,
                   messageId: Int,
                   key: ContactKey,
                   senderKey: ToxKey,
                   senderName: String,
                   message: String,
                   received: Boolean,
                   read: Boolean,
                   sent: Boolean,
                   timestamp: Timestamp,
                   size: Int,
                   `type`: MessageType,
                   fileKind: FileKind,
                   callEventKind: CallEventKind) {

  def logFormat(): Option[String] =
    if (!isFileTransfer) {
      val name =
        if (isMine) {
          ToxSingleton.tox.getName
        } else {
          senderName
        }

      val prettyTimestamp = TimestampUtils.prettyTimestamp(timestamp, isChat = true)

      Some(s"<$name> $message [$prettyTimestamp]")
    } else {
      None
    }

  def toNotificationFormat(context: Context): String = {
    `type` match {
      case MessageType.ACTION | MessageType.GROUP_ACTION =>
        s"$senderName $message"
      case MessageType.FILE_TRANSFER =>
        val extension = message.substring(message.lastIndexOf(".") + 1)
        if (FileUtils.imageExtensions.contains(extension)) {
          if (isMine) {
            context.getResources.getString(R.string.you_sent_image)
          } else {
            context.getResources.getString(R.string.friend_sent_image, senderName)
          }
        } else {
          if (isMine) {
            context.getResources.getString(R.string.you_sent_file)
          } else {
            context.getResources.getString(R.string.friend_sent_file, senderName)
          }
        }
      case MessageType.GROUP_MESSAGE =>
        s"$senderName: $message"

      case _ =>
        message
    }
  }

  def isMine: Boolean = {
    senderKey.equals(ToxSingleton.tox.getSelfKey)
  }

  def isFileTransfer: Boolean = {
    MessageType.transferValues.contains(`type`)
  }
}
